src/components/detail/Detail.vue

```diff
@@ -232,38 +112,37 @@ export default {
     },
 
     methods: {
-        filterChartTypes () {
-            const self = this
-
-            self.chartTypes = self.availableChartTypes.filter(function (c) {
-                if (!self.metricData) { return false; }
-                if (self.metricData.type === 'bars') { return c.chart !== 'line' }
-                if (self.metricData.type === 'lines') { return c.chart === 'line' }
-                return c.chart === 'table'
-            })
-
-            self.chartType = self.chartTypes[0].chart
-            self.chartIcon = self.chartTypes[0].icon
-        },
 
         wikiSelected (wiki) {
-            this.$emit('wiki', wiki)
+            this.$emit('wiki', wiki);
+            this.loadData();
         },
         loadData () {
-            const self = this
-
             this.highlightMetric = { name: this.metric, area: this.area }
 
-            config.metricData(this.metric, this.area).then(function (result) {
-                self.metricData = result
-                self.breakdowns = result.breakdowns
-            })
-
-            config.metrics(this.area).then(function (result) {
+            config.metricData(this.metric, this.area).then((result) => {
```

Comment:
Ok, thought more about how to manage this, let me know how you like this idea:

* loadData is renamed to configure, it gets the config and sets this.metricData and this.breakdowns.
* a computed called apiParameters depends on wiki, metricData and breakdowns (you can deep-depend btw, this is what you were looking for: https://vuejs.org/v2/api/#watch { deep: true }).  This returns what's passed to getData.
* a watcher watches apiParameters and calls this.loadData(this.apiParameters).
* loadData just calls the aqs api.